torch/_refs/linalg/utils.py

```diff
@@ -0,0 +1,42 @@
+import torch
+from torch._prims.utils import (
+    check,
+    get_higher_dtype,
+    is_half_dtype,
+    is_float_dtype,
+    is_complex_dtype,
+)
+
+from typing import Optional
+
+
+def check_fp_or_complex(dtype: torch.dtype, fn_name: str, half: bool = False):
+    check(
+        is_float_dtype(dtype) or is_complex_dtype(dtype),
+        lambda: f"{fn_name}: Expected a floating point or complex tensor as input. Got {dtype}",
+    )
+    check(
+        half or not is_half_dtype(dtype),
+        lambda: f"{fn_name}: Half precision dtypes not supported. Got {dtype}",
+    )
+
+
+def check_norm_dtype(dtype: Optional[torch.dtype], x_dtype: torch.dtype, fn_name: str):
```

Comment:
Comments for both of these functions describing what they do and how they're intended to be used